File: medium/78.子集/78.子集.js
/*
 * @lc app=leetcode.cn id=78 lang=javascript
 *
 * [78] 子集
 */
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
  const result = [[]]
  nums.forEach(n => {
    result.forEach(item => {
      result.push(item.concat(n))
    })
  })
  return result
}

File: medium/78.子集/78.子集.go
import "math"

/*
 * @lc app=leetcode.cn id=78 lang=golang
 *
 * [78] 子集
 */
func subsets(nums []int) [][]int {
	var ret [][]int
	ret = append(ret, []int{})
	size := len(nums)
	subsize := int(math.Pow(2, float64(size)))
	hash := 1
	for hash < subsize {
		var temp []int
		for k := 0; k < size; k++ {
			a := int(math.Pow(2, float64(k)))
			if (a & hash) != 0 {
				temp = append(temp, nums[k])
			}
		}
		ret = append(ret, temp)
		hash++
	}
	return ret
}

